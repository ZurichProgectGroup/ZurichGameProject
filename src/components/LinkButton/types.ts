import { ButtonHTMLAttributes, ReactNode } from 'react';

export type OwnProp = {
    className?: string,
    children: ReactNode,
    isButton?: boolean,
    to?: string
};

export type LinkProps = {
    to: string
    isButton?: false
} & OwnProp;

export type ButtonProps = {
    isButton: true,
} & ButtonHTMLAttributes<HTMLButtonElement> & OwnProp;

export type Props = LinkProps | ButtonProps;
