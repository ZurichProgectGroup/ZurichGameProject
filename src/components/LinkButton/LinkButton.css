.link-button {
    font-size: 14px;
    color: var(--link);
    text-transform: uppercase;
    text-shadow: 0 3px 10px rgba(var(--link-rgb), 0.4);
    text-decoration: underline;
    background: transparent;

    &:focus,
    &:hover {
        outline: none;
        opacity: 0.6;
    }
}
