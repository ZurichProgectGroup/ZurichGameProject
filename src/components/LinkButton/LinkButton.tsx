import React, { useMemo } from 'react';
import './LinkButton.css';
import cn from 'classnames';
import { Link } from 'react-router-dom';
import type { Props } from 'components/LinkButton/types';

const LinkButton = (props: Props) => {
    const {
        children,
        className,
        isButton = false,
        to = '',
        ...otherProps
    } = props;
    const classNames = useMemo(() => cn('link-button', className), [className]);

    if (isButton) {
        return (
            <button className={classNames} type="button" {...otherProps}>
                {children}
            </button>
        );
    }

    return (
        <Link className={cn('link-button', className)} to={to}>
            {children}
        </Link>
    );
};

export default LinkButton;
