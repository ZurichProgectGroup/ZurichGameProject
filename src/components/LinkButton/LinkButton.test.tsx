import React from 'react';
import { render } from '@testing-library/react';
import LinkButton from './LinkButton';

it('renders correctly', () => {
    const { asFragment } = render(<LinkButton isButton>test link</LinkButton>);

    expect(asFragment()).toMatchSnapshot();
});
