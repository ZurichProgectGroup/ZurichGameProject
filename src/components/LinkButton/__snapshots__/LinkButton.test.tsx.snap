// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class="link-button"
    type="button"
  >
    test link
  </button>
</DocumentFragment>
`;
